pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -827,7 +832,7 @@ func TestPodStatusChange(t *testing.T) {
 		for _, zone := range zones {
 			nodeController.zonePodEvictor[zone].Try(func(value scheduler.TimedValue) (bool, time.Duration) {
 				nodeUID, _ := value.UID.(string)
-				nodeutil.DeletePods(item.fakeNodeHandler, nodeController.recorder, value.Value, nodeUID, nodeController.daemonSetStore)
+				nodeutil.DeletePods(item.fakeNodeHandler, nodeController.recorder, value.Value, nodeUID, nodeController.tenantPartitionManagers[0].DaemonSetStore)
```

Comment:
why [0]?

Reply:
> why [0]?

It is a backwards-compatible test for In scale-up scenarios.

Because of the following code change in func startNodeLifecycleController() in cmd/kube-controller-manager/app/core.go, the tenantPartitionManagers will contain one member. So I use [0] to get the tenantPartitionManager.

if len(ctx.ComponentConfig.NodeLifecycleController.TenantServers) > 0 {
tpAccessors, err = nodeutil.GetTenantPartitionManagersFromServerNames(ctx.ComponentConfig.NodeLifecycleController.TenantServers, ctx.Stop)
} else {
tpAccessors, err = nodeutil.GetTenantPartitionManagersFromKubeClients([]clientset.Interface{kubeclient}, ctx.Stop)
}

Reply:
I see. thanks for the explanation. I'd suggest adding a comment about this as it's not straightforward to understand.